### PR TITLE
SIDM-3909 - IDAM prod2

### PR DESCRIPTION
**Pull request by henrydobson**
### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/SIDM-3909

### Change description ###

IDAM prod2

This deployment is for pre-switchover. IDAM service configuration and ingress will be updated on the go live date. Testing support will initially be enabled whilst the services are not available to the public for initial tests and then removed.

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[x] No
```

### CODE REVIEW

File: k8s/prod/common/idam/idam-web-public.yaml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    repository: https://hmctspublic.azurecr.io/helm/v1/repo/
+    name: idam-web-public
+    version: 0.2.3
+  values:
+    java:
+      image: hmctspublic.azurecr.io/idam/web-public:prod-1ebdebd1
+      ingressHost: hmcts-access.service.gov.uk
+      aadIdentityName: idam
+      environment:
+        STRATEGIC_SERVICE_URL: https://idam-api.platform.hmcts.net
```
Comment (kremi) [inline k8s/prod/common/idam/idam-web-public.yaml]: Can we use the `.internal` URLs? I don't feel comfortable having the live Prod URLs there from day 0.

Comment (henrydobson) [reply k8s/prod/common/idam/idam-web-public.yaml:25]: Platform URLs restored.

File: k8s/prod/common/idam/idam-api.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
+---
+apiVersion: helm.fluxcd.io/v1
+kind: HelmRelease
+metadata:
+  name: idam-api
+  namespace: idam
+  annotations:
+    flux.weave.works/automated: "true"
```
Comment (timja) [inline k8s/prod/common/idam/idam-api.yaml:8]: these annotations should use the new ones, search fluxcd in this repo

File: k8s/prod/common/idam/idam-api.yaml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+  chart:
+    repository: https://hmctspublic.azurecr.io/helm/v1/repo/
+    name: idam-api
+    version: 0.2.4
+  values:
+    java:
+      image: hmctspublic.azurecr.io/idam/api:latest
+      replicas: 2
+      ingressHost: idam-api-prod2.platform.hmcts.net
+      aadIdentityName: idam
```
Comment (timja) [inline k8s/prod/common/idam/idam-api.yaml:24]: this should be set in your chart

File: k8s/prod/common/idam/idam-api.yaml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      environment:
+        TESTING_SUPPORT_ENABLED: true
```
Comment (timja) [inline k8s/prod/common/idam/idam-api.yaml:26]: do you want this on in prod?

File: k8s/prod/common/idam/idam-web-admin.yaml [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+    retry: true
+  chart:
+    repository: https://hmctspublic.azurecr.io/helm/v1/repo/
+    name: idam-web-admin
+    version: 0.2.3
+  values:
+    java:
+      image: hmctspublic.azurecr.io/idam/web-admin:latest
+      ingressHost: idam-web-admin-prod2.platform.hmcts.net
+      aadIdentityName: idam
```
Comment (timja) [inline k8s/prod/common/idam/idam-web-admin.yaml:23]: this should be set in your chart

File: k8s/prod/common/idam/idam-web-public.yaml [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+    retry: true
+  chart:
+    repository: https://hmctspublic.azurecr.io/helm/v1/repo/
+    name: idam-web-public
+    version: 0.2.3
+  values:
+    java:
+      image: hmctspublic.azurecr.io/idam/web-public:latest
+      ingressHost: hmcts-access.service.gov.uk
+      aadIdentityName: idam
```
Comment (timja) [inline k8s/prod/common/idam/idam-web-public.yaml:23]: again in chart

File: k8s/prod/common/idam/idam-api.yaml (hunk 3 above)
Comment (henrydobson) [reply k8s/prod/common/idam/idam-api.yaml:24]: I will add it in the next revision.

File: k8s/prod/common/idam/idam-api.yaml (hunk 4 above)
Comment (henrydobson) [reply k8s/prod/common/idam/idam-api.yaml:26]: Whilst prod2 is deployed and not active, we plan to test the service and then remove the test data. Once this is complete, testing support will be disabled.

File: CODEOWNERS [hunk 7]
```diff
@@ -184,6 +184,9 @@ k8s/**/common/bar/identity.yaml @hmcts/Fees
 k8s/**/common/bar/namespace.yaml @hmcts/Fees
 
 # IDAM
+k8s/prod/common/idam/idam-api.yaml @hmcts/idam
+k8s/prod/common/idam/idam-web-public.yaml @hmcts/idam
+k8s/prod/common/idam/idam-web-admin.yaml @hmcts/idam
```
Comment (henrydobson) [inline CODEOWNERS:189]: Based on the readme, I think this is what you're after @timja

Comment (timja) [reply CODEOWNERS]: update the aat ones to be **

**Pull request merged**